### PR TITLE
KOTOR: Initial save/load game menu implementation

### PR DESCRIPTION
**Pull request by seedhartha**
Accesible from both main menu and ingame options menu.

### CODE REVIEW

File: src/engines/kotor/gui/saveload.h [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+#define ENGINES_KOTOR_GUI_SAVELOAD_H
+
+#include "src/engines/aurora/console.h"
+#include "src/engines/kotor/gui/gui.h"
+
+namespace Engines {
+
+namespace KotOR {
+
+enum : uint8 {
```
Comment (DrMcCoy) [inline src/engines/kotor/gui/saveload.h]: The typed enum is another C++11 thing we don't want, sorry. :P

**Comment by seedhartha** [comment]
Fixed compilation error. Also implemented saved games list loading. There is a bug with listbox item size which I am not yet sure how to workaround. What does DIMENSION field in GUI GFF file stand for?

File: src/common/filelist.h [hunk 2]
```diff
@@ -77,6 +77,11 @@ class FileList {
 	 */
 	bool addDirectory(const UString &directory, int recurseDepth = 0);
 
+	/** Add subdirectories of a directory to the list.
+	 *  @param directory directory to scan
+	 */
+	bool addSubDirectories(const UString &directory);
```
Comment (DrMcCoy) [inline src/common/filelist.h]: Could you separate adding this method into its own commit, please?

File: src/engines/aurora/gui.h [hunk 3]
```diff
@@ -75,8 +75,11 @@ class GUI : boost::noncopyable {
 	GUI *_sub; ///< The currently running sub GUI.
 
 
-	/** Add a widget. */
-	void addWidget(Widget *widget);
+	/**
+	 * Add a widget.
+	 * @param show show it immediately
+	 */
+	void addWidget(Widget *widget, bool show = false);
```
Comment (DrMcCoy) [inline src/engines/aurora/gui.h]: Changing this method should also be its own commit.

File: src/engines/jade/gui/main/main.cpp [hunk 4]
```diff
@@ -57,12 +57,12 @@ MainMenu::MainMenu(Module &module, ::Engines::Console *console) : ::Engines::Kot
...
-	addWidget(getListBox("ListBoxButtons")->createItem("MINIGAMES"));
-	addWidget(getListBox("ListBoxButtons")->createItem("OPTIONS"));
-	addWidget(getListBox("ListBoxButtons")->createItem("CREDITS"));
-	addWidget(getListBox("ListBoxButtons")->createItem("EXIT"));
+	addWidget(getListBox("ListBoxButtons")->createItem("NEW_GAME"), true);
+	addWidget(getListBox("ListBoxButtons")->createItem("LOAD_GAME"), true);
+	addWidget(getListBox("ListBoxButtons")->createItem("MINIGAMES"), true);
+	addWidget(getListBox("ListBoxButtons")->createItem("OPTIONS"), true);
+	addWidget(getListBox("ListBoxButtons")->createItem("CREDITS"), true);
+	addWidget(getListBox("ListBoxButtons")->createItem("EXIT"), true);
```
Comment (DrMcCoy) [inline src/engines/jade/gui/main/main.cpp]: Why? At that point, the GUI itself isn't shown yet.

Comment (seedhartha) [reply src/engines/jade/gui/main/main.cpp]: No idea, that how it worked before :)

Comment (DrMcCoy) [reply src/engines/jade/gui/main/main.cpp]: Huh, am I missing something? You added the parameter `show`, and added `if (show) widget->show()`. So the old behaviour would be the same as `addWidget(..., false);`, no?

Comment (seedhartha) [reply src/engines/jade/gui/main/main.cpp]: In *listbox.cpp*, function *createItem*, there was a non optional call to *show* on newly created item. It didn't work right for me, hence I moved that into *addWidget* function.

Comment (DrMcCoy) [reply src/engines/jade/gui/main/main.cpp]: Ah, okay :)

**Comment by seedhartha** [comment]
All done.

**Comment by seedhartha** [comment]
Done.

**Comment by DrMcCoy** [comment]
Merged as 2c50f7cc...b4ebfdf6, thanks! :)

**Comment by farmboy0** [comment]
What problem do you have with the DIMENSION?
